library/general/src/lib/cfa/sysctl.rb

```diff
@@ -118,39 +124,27 @@ def load
 
     # Writes sysctl configuration
     #
-    # Apart from writing the values to {PATH}, it updates the same entries in
-    # `/etc/sysctl.conf` to avoid confusion.
+    # Apart from writing the values to {PATH}, it cleans up the same entries in
+    # +/etc/sysctl.conf+ to avoid confusion.
     def save
       super
-      update_old_values
+      clean_old_values
     end
 
   private
 
     # Path to the agent to handle the +/etc/sysctl.conf+ file
     SYSCTL_AGENT_PATH = Yast::Path.new(".etc.sysctl_conf")
 
-    # Updates present values in `/etc/sysctl.conf` to reduce confusion
-    #
-    # @note Those values should be removed. However, it may cause the main comment to be removed
-    # too. So, for the time being, just update any parameter which is present with its new value.
-    def update_old_values
-      known_keys.each do |key|
-        old_value = Yast::SCR.Read(SYSCTL_AGENT_PATH + key)
-        next if old_value.nil? || old_value == data[key]
-
-        Yast::SCR.Write(SYSCTL_AGENT_PATH + key, data[key])
-      end
-      Yast::SCR.Write(SYSCTL_AGENT_PATH, nil)
-    end
-
-    # Returns the list of known attributes
-    #
-    # Just a helper method to get the list of defined attributes
-    #
-    # @return [String<Symbol>]
-    def known_keys
-      @known_keys ||= ATTRIBUTES.values.uniq
+    MAIN_SYSCTL_CONF_PATH = "/etc/sysctl.conf".freeze
+    KNOWN_KEYS_REGEXP = /^(#{KNOWN_KEYS.join("|")})/.freeze
```

Comment:
I think if we use regexp approach we need to ensure it does not fetch e.g. partial keys like net.ipv4.conf.all and net.ipv4.conf.all.forwarding. Another issue here is special regexp characters like `.` in keys which can result in unwanted match. In general using regexp for manipulating config files where we are not so sure how it can look like is dangerous (e.g. can be whitespace between start of line and key? ).
If you want cleanly remove some key I think that CFA approach is better. Just load directly that file and remove given key from it. It will keep comments.